### PR TITLE
chore(files_versions): improve and fix technical debt on `Version.vue`

### PR DESCRIPTION
**Pull request by skjnldsv**
After https://github.com/nextcloud/server/pull/43727/files#r1497933392

- Do not import some methods that were not designed for the same purpose
- Simplify code 
- Migrate to typescript
- Cleanup and added comments

### CODE REVIEW

File: apps/files_versions/src/components/Version.vue [hunk 1]
```diff
@@ -203,59 +208,53 @@ export default {
...
-		enableDeletion() {
+		enableDeletion(): boolean {
 			return this.capabilities.files.version_deletion === true
 		},
 
-		/** @return {boolean} */
-		hasDeletePermissions() {
-			return hasPermissions(this.fileInfo.permissions, Permission.DELETE)
+		hasDeletePermissions(): boolean {
+			return (this.fileInfo.permissions & Permission.DELETE) !== 0
```
Comment (artonge) [inline apps/files_versions/src/components/Version.vue:232]: I honestly don't understand how this is better, this kind of condition is famous for being hard to read and error-prone.

**Review by artonge — CHANGES_REQUESTED** [review]

File: apps/files_versions/src/components/Version.vue (hunk 1 above)
Comment (skjnldsv) [reply apps/files_versions/src/components/Version.vue:232]: We should refrain from importing methods that are from other apps

Comment (artonge) [reply apps/files_versions/src/components/Version.vue:232]: Then you can create a new function local to `files_versions` if you want, but the current change is not better in my opinion.

Comment (skjnldsv) [reply apps/files_versions/src/components/Version.vue:232]: If you prefer :+1:

**Review by artonge — APPROVED** [review]
Thanks for the upgrade :heart:

**Review by susnux — APPROVED** [review]
LGTM

**Pull request merged**